artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQSession.java

```diff
@@ -848,7 +848,7 @@ private ActiveMQMessageConsumer createConsumer(final ActiveMQDestination dest,
                   if (!response.isExists() || !response.getQueueNames().contains(AutoCreateUtil.getCoreQueueName(session, dest.getSimpleAddress()))) {
                      if (response.isAutoCreateQueues()) {
                         try {
-                           createQueue(dest, RoutingType.MULTICAST, dest.getSimpleAddress(), null, true, true, response);
+                           createQueue(dest, RoutingType.MULTICAST, dest.getSimpleAddress(), coreFilterString, true, true, response);
```

Comment:
This usecase could also use a test. Now that I think about it. Wouldn't it make sense to pair the filter with session, rather than the queue? In such case, the messages that don't fit the with the filter could still be consumed by a different client later on. Especially in case the queue haven't been created with any filter previously.

Reply:
The main reason to do it on the queue is specifically to avoid messages that 'werent asked for' originally for a topic-subscription-like-FQQN-queue (note this bit is specific to multicast/topic) from accumulating on the queue where they wont then be consumed (as e.g is happening for an AMQP case before this PR).

For real-topic-subscriptions comparison, JMS doesnt allow selectors to differ on a single sub (while sharing) and changing it later constitutes a new subscription beginning from that point without previous messages.